code/game/objects/effects/chem/chemsmoke.dm

```diff
@@ -150,6 +160,24 @@
 		else
 			message_admins("A chemical smoke reaction has taken place in ([whereLink]). No associated key.", 0, 1)
 			log_game("A chemical smoke reaction has taken place in ([where])[contained]. No associated key.")
+	else if (show_touch_log && touch_log_trigger)
+		var/mobnames = ""
+		if (touched_mobs.len > 1)
+			mobnames += "Affected players: "
+			var/i = 1
+			do
+				mobnames += "<A HREF='?_src_=holder;adminmoreinfo=\ref[touched_mobs[i]]'>?</a>"
+				if (touched_mobs[i+1])
+					mobnames += ", "
+			while (touched_mobs[i])
```

Comment:
I'm not seeing `i` being iterated here? It'll do an infinite amount of cycles if touched_mobs has 2 or more than 2 entries, as it'll always have `i` at a value of 1.